src/ui/MD/a.tsx

```diff
@@ -0,0 +1,21 @@
+import { Link } from "@mui/material";
+import React from "react";
+import { useNavigator } from "../React/Documentation";
+import { CorruptableText } from "../React/CorruptableText";
+import { Player } from "@player";
+
+export const isSpoiler = (title: string): boolean => title.startsWith("advanced/") && Player.sourceFileLvl(1) === 0;
+
+export const A = (props: React.PropsWithChildren<{ href?: string }>): React.ReactElement => {
+  const navigator = useNavigator();
+  const ref = props.href ?? "";
+  if (ref.startsWith("http"))
+    return (
+      <Link rel="noopener noreferrer" href={props.href} target="_blank">
+        {props.children}
+      </Link>
+    );
+
+  if (isSpoiler(ref)) return <CorruptableText content="UNKNOWN" />;
+  return <Link onClick={() => navigator.navigate(ref)}>{props.children}</Link>;
```

Comment:
```suggestion
  return (
    <Link component={"button"} variant={"body2"} onClick={() => navigator.navigate(ref)}>
      {props.children}
    </Link>
  );
```
Links with no href are inaccessible. They don't have the right cursor, are not focusable, and even if we add href=# to make them focusable, they don't actually run the onClick when they are activated with keyboard. With functionality on an onClick, we need to use a button element, change is based on guidance from mui: https://mui.com/material-ui/react-link/#accessibility

Reply:
I agree that something should change here, but I disagree that we should be changing to buttons. The action here *is* navigation, therefore they ought to be links.

Instead, I think these should have real hrefs, and we should be making our router a bit smarter so that it works with them. Anything that has a history should potentially use the browser's history.

This page has a good overview of the concepts: <https://blog.pshrmn.com/how-single-page-applications-work/>

Reply:
Oh, another thing on this thread: There are some people (like me) who instinctively ctrl-click to open all documentation links like this in new tabs. Having these be actual links is good because it supports this... theoretically. However, since it would be opening the game to render the docs, you'd rapidly end up with 10 copies of the game running the same save, which is... not ideal.

Probably the best would be if the doc links opened to an external md renderer (such as github) if opened externally. Actually, the more I think about it, the more important that is: I do *not* want to be looking at these in-game. (It's just how I'm wired, sorry)